tensorflow/compiler/jit/xla_compilation_cache.h

```diff
@@ -163,6 +173,21 @@ class XlaCompilationCache : public ResourceBase {
     std::unique_ptr<xla::LocalExecutable> executable TF_GUARDED_BY(mu);
   };
 
+  Status CompileStrict(
+    Entry* entry, const XlaCompiler::Options& options,
+    const std::vector<XlaCompiler::Argument>& args,
+    const string &function_name,
+    const std::function<Status(XlaCompiler* compiler,
+                               const std::vector<XlaCompiler::Argument>& args,
+                               XlaCompiler::CompilationResult*)>& compile_fn);
+  Status CompileAsynchronous(
+    Entry* entry, const XlaCompiler::Options& options,
+    const std::vector<XlaCompiler::Argument>& args,
+    const string &function_name,
+    const std::function<Status(XlaCompiler* compiler,
+                               const std::vector<XlaCompiler::Argument>& args,
+                               XlaCompiler::CompilationResult*)>& compile_fn);
```

Comment:
Have you looked into making `CompileAsynchronous` a separate method / class that lives outside `XlaCompilationCache` and interacts with it only via its public interface (suitably refactored)?  That might simplify the change a bit.

Reply:
I did not.
Looking at it, I'm not sure it fits, unless the refactoring is substantial.
Currently CompileImpl and ComplieStrict are both XlaCompilationCache members, as is Entry. CompileAsynchronus sits in between these two functions, and operates on the Entry Type.

Reply:
I did not.
Looking at it, I'm not sure it fits, unless the refactoring is substantial.
Currently CompileImpl and ComplieStrict are both XlaCompilationCache members, as is Entry. CompileAsynchronus sits in between these two functions, and operates on the Entry Type.